heat/core/linalg/basics.py

```diff
@@ -44,6 +45,118 @@
 ]
 
 
+def cholesky(x: DNDarray, upper: bool = False) -> DNDarray:
+    """
+    Returns the Cholesky decomposition of a Hermitian or symmetric positive-definite matrix.
+
+    Parameters
+    ----------
+    x : DNDarray
+        Symmetric or Hermitian positive-definite matrix. Shape = (...,M,M)
+    upper : bool, optional
+        Whether a lower or upper triangular matrix is returned.
+
+    Note
+    ----
+    Distributed matrices with complex valued entries are currently not supported.
+
+    Examples
+    --------
+    >>> A = ht.array([[1,-2j],[2j,5]])
+    >>> A
+    DNDarray([[ (1+0j), (-0-2j)],
+              [     2j,  (5+0j)]], dtype=ht.complex64, device=cpu:0, split=None)
+    >>> L = ht.linalg.cholesky(A)
+    >>> L
+    DNDarray([[(1+0j),     0j],
+              [    2j, (1+0j)]], dtype=ht.complex64, device=cpu:0, split=None)
+    >>> L @ L.T.conj()
+    DNDarray([[(1+0j),    -2j],
+              [    2j, (5+0j)]], dtype=ht.complex64, device=cpu:0, split=None)
+    """
+    if x.ndim < 2:
+        raise RuntimeError("DNDarray must be at least two-dimensional.")
+
+    m, n = x.shape[-2:]
+
+    if m != n:
+        raise RuntimeError("Last two dimensions of the DNDarray must be square.")
```

Comment:
How about "Expected a DNDarray of shape (..., M, M), got ..." or "Last two dimensions of DNDarray must be equal, are ..."